core/src/main/scala/org/apache/spark/deploy/history/FsHistoryProvider.scala

```diff
@@ -187,47 +201,78 @@ private[history] class FsHistoryProvider(conf: SparkConf) extends ApplicationHis
         }
         .flatMap { entry =>
           try {
-            Some(replay(entry, new ReplayListenerBus()))
+            Some((getModificationTime(entry).get, entry))
           } catch {
             case e: Exception =>
               logError(s"Failed to load application log data from $entry.", e)
               None
           }
         }
-        .sortWith(compareAppInfo)
+        .sortWith(_._1 >= _._1)
+        .map(_._2)
+
+      val itor = logInfos.iterator
```

Comment:
This is probably more cleany expressed as:

    logInfos.sliding(20, 20).foreach {
      replayExecutor.submit(...)
    }